src/main/java/julia/git/Directory.java

```diff
@@ -0,0 +1,69 @@
+package julia.git;
+
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+public class Directory implements Listable {
+    private final HashMap<String, Listable> content = new HashMap<>();
+
+    public void add(String name, Listable list) {
+        content.put(name, list);
+    }
+
+    public List<String> list() {
+        return new ArrayList<>(content.keySet());
+    }
+
+    public Listable get(String name) {
+        return content.get(name);
+    }
+
+    @Override
+    public boolean isDirectory() {
+        return true;
+    }
+
+    @Override
+    public String getContent() throws UnsupportedEncodingException, NoSuchAlgorithmException {
```

Comment:
Outside world shouldn't know about these exceptions. Let's catch them right in the method and re-throw as `RuntimException`